Lexc improvements

elmurod1202 opened the pull request.

elmurod1202 committed:
ADD: NP-ORG-COMPOUND-PX (Part 1)

elmurod1202 committed:
ADD: NP-ORG-COMPOUND-PX (Part 2)

elmurod1202 committed:
ADD: NP-ORG-COMPOUND-PX (Part 3, Final part)

elmurod1202 committed:
Fix: SETimes

elmurod1202 committed:
Hitparade

elmurod1202 committed:
ADD: Roman Numerals

elmurod1202 committed:
Add: Words from frequency list

elmurod1202 committed:
ADD: NP-ANT-M, Letter(s): A-D

elmurod1202 committed:
ADD: NP-ANT-M, Letter(s): E-K

elmurod1202 committed:
ADD: NP-ANT-M, Letter(s): E-N

elmurod1202 committed:
ADD: NP-ANT-M, Letter(s): O-T

elmurod1202 committed:
ADD: NP-ANT-M, Letter(s): U-Z

elmurod1202 committed:
ADD some words

elmurod1202 committed:
Add: New NP-ANT-M entries

elmurod1202 committed:
Add: Some more NP-ANT-M entries

elmurod1202 committed:
Minor fix

elmurod1202 committed:
Add: NP-ANT-F, letters: A-M

elmurod1202 committed:
Add: NP-ANT-F, letters: N-Z

elmurod1202 committed:
Fix: lowercase names of monthes

elmurod1202 committed:
Massive inclusion of Cognomens, Part 1/4

elmurod1202 committed:
Massive inclusion of Cognomens, Part 2/4

elmurod1202 committed:
Massive inclusion of Cognomens, Part 3/4

elmurod1202 committed:
Massive inclusion of Cognomens, Part 4/4

elmurod1202 committed:
Add: Some new NP-TOPs

elmurod1202 committed:
Some new NP-ORGs

elmurod1202 committed:
Add: NP-COG-ZADEs

elmurod1202 committed:
Add: NP-COG-OVs

elmurod1202 committed:
Add: NP-PAT-VICHs

elmurod1202 committed:
uyushgan jinoyatchilik

elmurod1202 committed:
Fix: removed "%%-"s from lexc.

elmurod1202 committed:
add: uyushtir

elmurod1202 committed:
Add: Frequency words

elmurod1202 committed:
Add: Frequent words

elmurod1202 committed:
Fix: typo

elmurod1202 committed:
Add: NP-AL

elmurod1202 committed:
Fix: typo

jonorthwash commented on apertium-uzb.uzb.lexc:
```diff
@@ -40745,3 +40800,5557 @@ Amerika% Qoʻshma% Shtatlari:Amerika% Qoʻshma% Shtatlari  NP-TOP ; ! "" ! El++
+Acimambetov:Acimambetov  NP-COG-MF ; ! "" ! El++
+Acimambetova:Acimambetova  NP-COG-MF ; ! "" ! El++
+Acimovich:Acimovich  NP-COG-MF ; ! "" ! El++
+Adagamov:Adagamov  NP-COG-MF ; ! "" ! El++
```
There's a special lexicon for names ending in -ov and -ova in other Turkic transducers (check e.g. kaz and tat).  Could you copy that way of doing this?  Same for -vich, -ovna, etc.

elmurod1202 replied:
Good, I was thinking of having a separate lexicons for that, gonna copy from kaz(or tat) then. For now, I just wanted to have the same tag as in tur.lexc, later regroup again.  I also am planning to import ANT-M and ANT-Fs from Uzbek names book.

jonorthwash commented on apertium-uzb.uzb.lexc:
```diff
@@ -40745,3 +40800,5557 @@ Amerika% Qoʻshma% Shtatlari:Amerika% Qoʻshma% Shtatlari  NP-TOP ; ! "" ! El++
+Twitter:Twitter  NP-AL ; ! "" ! El++
+Ukrenergo:Ukrenergo  NP-AL ; ! "" ! El++
+Ukrinform:Ukrinform  NP-AL ; ! "" ! El++
+Ukrtelekom:Ukrtelekom  NP-AL ; ! "" ! El++
```
Many of these can be `<np><org>`—better than NP-AL.

elmurod1202 replied:
Sure, will sort them out again.

elmurod1202 replied:
Sorted them out, they They became NP-ALs, NP-ORGs and an NP-TOP both in tur.lexc, tur-uzb.dix and respective translations in uzb.lexc as well.

jonorthwash commented on apertium-uzb.uzb.lexc:
```diff
@@ -3392,12 +3397,12 @@ bariton:bariton N1 ; !
 barkash:barkash N1 ; ! 
-barqaror:barqaror A1 ; ! 
-barqarorlik:barqarorlik N1 ;  ! 
+barqaror:barqaror A1 ; ! "stabil" 
```
"stable"

elmurod1202 replied:
Hah, looks like even-though I was born after the Uzbekistan's independence, deep down inside, I have some Russian instincts :D . Fixed it.

jonorthwash commented on apertium-uzb.uzb.lexc:
```diff
@@ -35708,7 +35740,7 @@ elfurush:elfurush N1 ; ! 0.9999999987189747
-elchi:elchi N1 ; ! "postman"
+elchi:elchi N1 ; ! "ambassador"
 elchi:Elchi N1 ; ! "ambassador"
 Elchi:Elchi N1 ; ! "ambassador"
```
Why do we need three versions of this?  Apertium should handle capitalisation differences automatically.

elmurod1202 replied:
Does it handle well?. Ex: names of the months in Uzbek were already there in capital letters, but I saw all month names in lowercases not-analysed when calculating uzb hfst coverage. So I was second-guessing about it.

jonorthwash replied:
It only handles it well if you put the default form in lower case. Uppercase forms should only be added to the dictionary when that's the only correct way of writing a word (like names).

elmurod1202 replied:
Noted. I fixed this and entries alike this one

jonorthwash commented on apertium-uzb.uzb.lexc:
```diff
@@ -35900,6 +35932,7 @@ erkatoy:erkatoy N1 ; ! 0.9999965658440407
 erka:erka N1 ; ! 0.9995437488829819
 erkinlik:erkinlik N1 ; ! 0.999999999995661
 erkin:erkin A1 ; ! "free"
+erkin:erkin ADV ; ! "freely"
```
This isn't covered by `<adj><advl>` readings?

elmurod1202 replied:
Apparently, it is. So it is not really necessary to add them separately?. I mean, yes when analysing uzbek text, it is handled, but is it also handled properly when there is an entry at bidix: `<l>erken("adv")</l>  <r>erkin("adv")</r>`?

jonorthwash replied:
For translation, use `erkin<adj><advl>` too.

elmurod1202 replied:
I guess I got a grasp of this now (looking at examples from kaz-tat), fixed it.

elmurod1202 committed:
stabil

elmurod1202 committed:
Merge branch with upstream

elmurod1202 committed:
Added to .gitignore: /tests/morphotactics/.deps/*

elmurod1202 committed:
Uzbzek names book: 10191 Male names

elmurod1202 committed:
Uzbek names book: 5117 Female names

jonorthwash commented:
The new names seem to have extra spaces.  Did you check that it compiles before committing?

elmurod1202 committed:
Fix: Removed spaces around colon mark in Uzbek names.

elmurod1202 commented:
> The new names seem to have extra spaces. Did you check that it compiles before committing?

Oh, my bad, just fixed it. I was quick to push changes as soon as it compiled without an error, not noticing the spaces.

elmurod1202 committed:
Some more names from names book(Source 1)

elmurod1202 committed:
Fix: problem with spaces

elmurod1202 committed:
Bulk action: Change "‘"(U2018) to "ʻ"(U02BB)

elmurod1202 committed:
Add: More Uzbek names (Source 2)

elmurod1202 committed:
Bulk action: Fix all "Tutuq belgisi"(glottal stop) --> U02BC

elmurod1202 committed:
Little thing

elmurod1202 committed:
Add: Uzbek male names (Source 3)

elmurod1202 committed:
Add: Uzbek female names (Source 3)

elmurod1202 committed:
Merge elmurod1202/apertium-uzb with apertium/apertium-uzb

elmurod1202 committed:
Fix: some adverbs are covered by <adj><advl>s

elmurod1202 committed:
Fix: got rid of Capitalisation issue

elmurod1202 committed:
bu yerda

elmurod1202 committed:
Removing all NP-AL entries to sort out and put back

elmurod1202 committed:
Putting back NP-AL entries after sort-out, Part: 1/2

elmurod1202 committed:
Putting back NP-AL entries after sort-out, Part: 2/2

jonorthwash commented:
I'm curious about these forms; what are they?
```
{+ad:ad NP-AL ; !+}
{+al:al NP-AL ; !+}
{+am:am NP-AL ; !+}
{+bei:bei NP-AL ; !+
{+ibn:ibn NP-AL ; !+}
{+la:la NP-AL ; !+}
{+le:le NP-AL ; !+}
{+les:les NP-AL ; !+}
{+lès:lès NP-AL ; !+}
```

jonorthwash commented:
> I'm curious about these forms; what are they?
> 
> ```
> {+ad:ad NP-AL ; !+}
> {+al:al NP-AL ; !+}
> {+am:am NP-AL ; !+}
> {+bei:bei NP-AL ; !+
> {+ibn:ibn NP-AL ; !+}
> {+la:la NP-AL ; !+}
> {+le:le NP-AL ; !+}
> {+les:les NP-AL ; !+}
> {+lès:lès NP-AL ; !+}
> ```

Oh, nevermind, these are forms you deleted, right?

jonorthwash commented:
Why did you move to `U02BC`?  Uzbek Wikipedia uses `U02BB`, which I thought was the standard?  (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)

jonorthwash commented:
The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.

jonorthwash commented:
Forms like this should be able to be generated automatically:
```
Chernishov:Chernishov  NP-COG-MF ; ! "" ! El++
Chernishova:Chernishova  NP-COG-MF ; ! "" ! El++
```
Forms like this appear to be patronymics, which can also be generated for `<m>` and `<f>` forms automatically:
```
Cholakovich:Cholakovich  NP-COG-MF ; ! "" ! El++
```

jonorthwash reviewed (changes requested):
See above comments

elmurod1202 commented:
> I'm curious about these forms; what are they?
> 
> ```
> {+ad:ad NP-AL ; !+}
> {+al:al NP-AL ; !+}
> {+am:am NP-AL ; !+}
> {+bei:bei NP-AL ; !+
> {+ibn:ibn NP-AL ; !+}
> {+la:la NP-AL ; !+}
> {+le:le NP-AL ; !+}
> {+les:les NP-AL ; !+}
> {+lès:lès NP-AL ; !+}
> ```

These would be parts of NP-ORGS, no?. Why I added them is because, first of all, they exist in turkish.lexc(and all the time I consider tur.lexc as an ideal package and try to replicate what's there), second of all, they help improve the coverage. Was it wrong?(If so, I'll fix them from both Turkish and Uzbek)

elmurod1202 commented:
> Why did you move to `U02BC`? Uzbek Wikipedia uses `U02BB`, which I thought was the standard? (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)

The shortest answer is: I didn't. Explanation: The apostrophe in <oʻ> and <gʻ> are U02BB and I converted all to that, there is another apostrophe in the alphabet: "tutuq belgisi"(phonetic glottal stop, <ъ> in cyrillic script) and itʻs U02BC,(Ex: aʼlo, maʼno, eʼzoz) I fixed them as well.

elmurod1202 commented:
> The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.

This way of announcing roman numerals was also copied from apertium-tur. Now I see what you(that was you, wasn't?) did in apertium-kaz/tat do deal with it (<(M | D | C | L | X | V | I)+> NUM-ROMAN ;). That looks more appropriate, I'll fix both tur and uzb if you confirm.

jonorthwash commented:
> > I'm curious about these forms; what are they?
> > ```
> > {+ad:ad NP-AL ; !+}
> > {+al:al NP-AL ; !+}
> > {+am:am NP-AL ; !+}
> > {+bei:bei NP-AL ; !+
> > {+ibn:ibn NP-AL ; !+}
> > {+la:la NP-AL ; !+}
> > {+le:le NP-AL ; !+}
> > {+les:les NP-AL ; !+}
> > {+lès:lès NP-AL ; !+}
> > ```
> 
> These would be parts of NP-ORGS, no?. Why I added them is because, first of all, they exist in turkish.lexc(and all the time I consider tur.lexc as an ideal package and try to replicate what's there), second of all, they help improve the coverage. Was it wrong?(If so, I'll fix them from both Turkish and Uzbek)

1. I'm not sure it's a good idea to always copy other Apertium pairs.  `tur.lexc` is generally in better shape than `uzb.lexc`, but it's far from perfect.

2. I'm not sure if we need these forms, but it might not hurt.  I don't know.  Let's not worry about it now, maybe...

jonorthwash commented:
> > Why did you move to `U02BC`? Uzbek Wikipedia uses `U02BB`, which I thought was the standard? (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)
> 
> The shortest answer is: I didn't. Explanation: The apostrophe in <oʻ> and <gʻ> are U02BB and I converted all to that, there is another apostrophe in the alphabet: "tutuq belgisi"(phonetic glottal stop, <ъ> in cyrillic script) and itʻs U02BC,(Ex: aʼlo, maʼno, eʼzoz) I fixed them as well.

Oh!  I didn't know these two were meant to be encoded differently!  So, how do we know what the right encoding of them is?

jonorthwash commented:
> > The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.
> 
> This way of announcing roman numerals was also copied from apertium-tur. Now I see what you(that was you, wasn't?) did in apertium-kaz/tat do deal with it (<(M | D | C | L | X | V | I)+> NUM-ROMAN ;). That looks more appropriate, I'll fix both tur and uzb if you confirm.

That was probably @IlnarSelimcan, and I assume that's right and should be okay to copy.

elmurod1202 commented:
> > > Why did you move to `U02BC`? Uzbek Wikipedia uses `U02BB`, which I thought was the standard? (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)
> > 
> > 
> > The shortest answer is: I didn't. Explanation: The apostrophe in <oʻ> and <gʻ> are U02BB and I converted all to that, there is another apostrophe in the alphabet: "tutuq belgisi"(phonetic glottal stop, <ъ> in cyrillic script) and itʻs U02BC,(Ex: aʼlo, maʼno, eʼzoz) I fixed them as well.
> 
> Oh! I didn't know these two were meant to be encoded differently! So, how do we know what the right encoding of them is?

Wiki page on "Uzbek alphabet" says that U02BB is an apostrophe used for <oʻ> and <gʻ>, while U02BC is used for "tutuq belgisi"(aka glottal stop?).  I fixed all apostrophes in uzb.lexc in my branch to what they are supposed to be. Yet there is a problem still standing is that due to the lack of proper keyboard layout for Uzbek alphabet, those apostrophes appear in varieties of forms, it has to be solved though (Issue #2).

ftyers commented:
> > > The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.
> > 
> > 
> > This way of announcing roman numerals was also copied from apertium-tur. Now I see what you(that was you, wasn't?) did in apertium-kaz/tat do deal with it (<(M | D | C | L | X | V | I)+> NUM-ROMAN ;). That looks more appropriate, I'll fix both tur and uzb if you confirm.
> 
> That was probably @IlnarSelimcan, and I assume that's right and should be okay to copy.

This might be problematic for the "entry beginning with whitespace" bug. You might want to use `[ ... ]` instead of `( ... )`.

elmurod1202 committed:
Fix: Apostrophes(-> U02BB)

elmurod1202 committed:
Fix: Glottal stop Apostrophes(-> U02BC)

elmurod1202 committed:
Fix: Better handling of Roman numerals

elmurod1202 committed:
Bug fix o roman numerals

elmurod1202 committed:
Add: Dress Sizes
